### PR TITLE
Sa core types and carton details form fixes

### PR DESCRIPTION
**Pull request by lewisrenfrew**
- only redirect to view screen of a newly created entity when 'RECEIVE_NEW...' action is dispatched
- enact redirect via middleware

### CODE REVIEW

File: src/Service.Host/client/src/components/CartonType.js [hunk 1]
```diff
@@ -32,24 +31,6 @@ function CartonType({
     const editing = () => editStatus === 'edit';
     const viewing = () => editStatus === 'view';
 
-    useEffect(() => {
-        if (!creating()) {
-            initialise({ cartonTypeId });
-        }
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/CartonType.js]: initialise logic moved to the HOC

File: src/Service.Host/client/src/components/CartonType.js [hunk 2]
```diff
@@ -32,24 +31,6 @@ function CartonType({
...
-    }, [cartonTypeId]);
-
-    useEffect(() => {
-        if (creating() && errorMessage) {
-            history.push(`/products/maint/carton-types/create`);
-        }
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/CartonType.js]: redirects moved to middleware

File: src/Service.Host/client/src/components/CartonType.js [hunk 3]
```diff
@@ -142,7 +122,6 @@ function CartonType({
                                 label="Description"
                                 fullWidth
                                 onChange={handleFieldChange}
-                                maxLength={50}
```
Comment (atcampbell) [inline src/Service.Host/client/src/components/CartonType.js]: Why did you need to remove these?

File: src/Service.Host/client/src/components/CartonType.js [hunk 4]
```diff
@@ -90,9 +68,12 @@ function CartonType({
     };
 
     const handleFieldChange = (propertyName, newValue) => {
-        setEditStatus('edit');
+        if (editStatus === 'view') {
+            setEditStatus('edit');
+        }
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/CartonType.js]: Don't set edit status if already editing / creating

File: src/Service.Host/client/src/components/CartonType.js (hunk 3 above)
Comment (lewisrenfrew) [reply src/Service.Host/client/src/components/CartonType.js]: hmmm.. I have no memory of removing that line. Will re-add before i get round to merging.
